### ISSUE TITLE
Unable to connect to openvpn with the new openvpn-gui version

### ISSUE THREAD

**Issue by Guigz25** [OpenVPN/openvpn#249]
**Describe the bug**
Unable to connect to OpenVPN after the openvpn-gui 2.6.0 update.
It works perfectly fine in 2.5.8

**To Reproduce**
Here is the client config file : 

```
client
dev tun
proto tcp

remote x.x.x.x 443
remote x.x.x.x 443

resolv-retry infinite

ca ca
cert xxxxxxxxxx
key xxxxxxxxxxx.key

remote-cert-tls server
tls-auth xxxxxxxxxxxxx 1

replay-persist file
data-ciphers AES-256-GCM:AES-128-GCM
cipher AES-256-GCM
auth SHA256

verb 3

route x.x.x.x x.x.x.x net_gateway
route x.x.x.x x.x.x.x net_gateway
route x.x.x.x x.x.x.x net_gateway
```

Here is the error in the client log file : 
```
2023-02-13 16:53:28 OpenVPN 2.6.0 [git:v2.6.0/b999466418dddb89] Windows-MSVC [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [AEAD] [DCO] built on Feb  6 2023
2023-02-13 16:53:28 Windows version 10.0 (Windows 10 or greater), amd64 executable
2023-02-13 16:53:28 library versions: OpenSSL 3.0.7 1 Nov 2022, LZO 2.10
2023-02-13 16:53:28 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:25341
2023-02-13 16:53:28 Need hold release from management interface, waiting...
2023-02-13 16:53:28 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:59296
2023-02-13 16:53:28 MANAGEMENT: CMD 'state on'
2023-02-13 16:53:28 MANAGEMENT: CMD 'log on all'
2023-02-13 16:53:28 MANAGEMENT: CMD 'echo on all'
2023-02-13 16:53:28 MANAGEMENT: CMD 'bytecount 5'
2023-02-13 16:53:28 MANAGEMENT: CMD 'state'
2023-02-13 16:53:28 MANAGEMENT: CMD 'hold off'
2023-02-13 16:53:28 MANAGEMENT: CMD 'hold release'
2023-02-13 16:53:32 MANAGEMENT: CMD 'password [...]'
2023-02-13 16:53:32 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
2023-02-13 16:53:32 Outgoing Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-13 16:53:32 Incoming Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-13 16:53:32 TCP/UDP: Preserving recently used remote address: [AF_INET]10.210.152.3:443
2023-02-13 16:53:32 ovpn-dco device [OpenVPN Data Channel Offload] opened
2023-02-13 16:53:32 TCP_CLIENT link local: (not bound)
2023-02-13 16:53:32 TCP_CLIENT link remote: [AF_INET]10.210.152.3:443
2023-02-13 16:53:32 MANAGEMENT: >STATE:1676303612,WAIT,,,,,,
2023-02-13 16:53:32 MANAGEMENT: >STATE:1676303612,AUTH,,,,,,
2023-02-13 16:53:32 TLS: Initial packet from [AF_INET]10.210.152.3:443, sid=d45af02a 24bdfe31
2023-02-13 16:53:32 VERIFY OK: depth=2, C=XX, ST=XXXXXX, L=XXXXX, O=XXXXXX, OU=XXXXX, CN=RootCA, emailAddress=XXXXXXXXX
2023-02-13 16:53:32 VERIFY OK: depth=1, C=XX, ST=XXXXXX, L=XXXXX, O=XXXXXX, OU=XXXXX, CN=ServerIntermediateCA, emailAddress=XXXXXXXXX
2023-02-13 16:53:32 VERIFY KU OK
2023-02-13 16:53:32 Validating certificate extended key usage
2023-02-13 16:53:32 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2023-02-13 16:53:32 VERIFY EKU OK
2023-02-13 16:53:32 VERIFY OK: depth=0, C=FR, ST=Rhone Alples, L=Grenoble, O=SPIE, OU=ICS CSEGC=XX, ST=XXXXXX, L=XXXXX, O=XXXXXX, OU=XXXXX, CN=OpenVPN1-server, emailAddress=XXXXXXX
2023-02-13 16:53:32 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 2048 bit RSA, signature: RSA-SHA256
2023-02-13 16:53:32 [OpenVPN1-server] Peer Connection Initiated with [AF_INET]10.210.152.3:443
2023-02-13 16:53:32 TLS: move_session: dest=TM_ACTIVE src=TM_INITIAL reinit_src=1
2023-02-13 16:53:32 TLS: tls_multi_process: initial untrusted session promoted to trusted
2023-02-13 16:53:32 PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS x.x.x.x,dhcp-option DNS x.x.x.x,dhcp-option DOMAIN prd.gre.loc,route x.x.x.x 255.240.0.0,route x.x.x.x 255.0.0.0,route x.x.x.x 255.255.0.0,route x.x.x.x 255.255.0.0,route-gateway x.x.x.x,topology subnet,ping 10,ping-restart 60,ifconfig x.x.x.x 255.255.255.0,peer-id 0,cipher AES-256-GCM'
2023-02-13 16:53:32 OPTIONS IMPORT: timers and/or timeouts modified
2023-02-13 16:53:32 OPTIONS IMPORT: --ifconfig/up options modified
2023-02-13 16:53:32 OPTIONS IMPORT: route options modified
2023-02-13 16:53:32 OPTIONS IMPORT: route-related options modified
2023-02-13 16:53:32 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
2023-02-13 16:53:32 OPTIONS IMPORT: peer-id set
2023-02-13 16:53:32 OPTIONS IMPORT: data channel crypto options modified
2023-02-13 16:53:33 interactive service msg_channel=636
2023-02-13 16:53:33 MANAGEMENT: >STATE:1676303613,ASSIGN_IP,,10.8.1.2,,,,
2023-02-13 16:53:33 INET address service: add 10.8.1.2/24
2023-02-13 16:53:33 TUN: adding address failed using service: Élément introuvable.   [status=1168 if_index=61]
2023-02-13 16:53:34 IPv4 dns servers set using service
2023-02-13 16:53:34 DNS domain set using service
2023-02-13 16:53:34 TUN: setting IPv4 mtu using service failed: Élément introuvable.   [status=1168 if_index=61]
2023-02-13 16:53:34 MANAGEMENT: >STATE:1676303614,ADD_ROUTES,,,,,,
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x x.x.x.x METRIC 200
2023-02-13 16:53:34 Route addition via service succeeded
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x x.x.x.x METRIC 200
2023-02-13 16:53:34 Route addition via service succeeded
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x x.x.x.x METRIC 200
2023-02-13 16:53:34 Route addition via service succeeded
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x 10.8.1.1 METRIC 200
2023-02-13 16:53:34 Warning: route gateway is not reachable on any active network adapters: 10.8.1.1
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x 10.8.1.1 METRIC 200
2023-02-13 16:53:34 Warning: route gateway is not reachable on any active network adapters: 10.8.1.1
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x 10.8.1.1 METRIC 200
2023-02-13 16:53:34 Warning: route gateway is not reachable on any active network adapters: 10.8.1.1
2023-02-13 16:53:34 C:\windows\system32\route.exe ADD x.x.x.x MASK x.x.x.x 10.8.1.1 METRIC 200
2023-02-13 16:53:34 Warning: route gateway is not reachable on any active network adapters: 10.8.1.1
2023-02-13 16:53:34 Initialization Sequence Completed
2023-02-13 16:53:34 MANAGEMENT: >STATE:1676303614,CONNECTED,ROUTE_ERROR,10.8.1.2,x.x.x.x,443,,

```

**Expected behavior**
Be able to be connected to OpenVPN with the 2.6.0

**Version information (please complete the following information):**
 - OS serveur:  Ubuntu 22.04
 - OS client : Windows 10 Pro
 - OpenVPN version: 2.6.0

**Comment by Guigz25** [comment]
> 

Thanks for you answer. 
The openvpn server is in Ubuntu 22.04. The client is in Windows 10. 
I've updated the log file.

**Comment by selvanair** [comment]
> 2023-02-13 16:53:32 ovpn-dco device [OpenVPN Data Channel Offload] opened
...
> 2023-02-13 16:53:33 TUN: adding address failed using service: Élément introuvable. [status=1168 if_index=61]

"Element not found"

Looks like the DCO adapter is not functional. Could this be some incompatibility with another installed driver -- like in https://github.com/OpenVPN/ovpn-dco-win/issues/31

**Comment by lstipakov** [comment]
Can you post your adapter settings to see if you have some 3rd party filter driver installed? Like this
![Sieppaa2](https://user-images.githubusercontent.com/643110/218521529-4ee0c047-8d2b-40c5-a0de-b357e92e954a.JPG)
![Sieppaa](https://user-images.githubusercontent.com/643110/218521540-bd7276d3-c6c1-455f-ad87-69f20afbd302.JPG)

**Comment by Guigz25** [comment]
Hi, 
So I managed to get the VPN working by deactivating the option DNE LightWeight Filter on the "OpenVPN Data Channel Offload" adapter : 

![image](https://user-images.githubusercontent.com/56647449/218978289-ee36fdaa-b9a7-4597-bef3-3e2375b9c4ff.png)

![image](https://user-images.githubusercontent.com/56647449/218979156-393a59ac-b2a4-4be8-97b1-851a9a66667b.png)

Is it an expected behavior ? 
What is precisely the DNE LightWeight Filter option ? 

Thanks in advance, 
Best Regards

**Comment by Guigz25** [comment]
Ok so I'm connected but after a few sec, I'm disconnected. 

Here is the log file : 
`
2023-02-15 09:55:41 OpenVPN 2.6.0 [git:v2.6.0/b999466418dddb89] Windows-MSVC [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [AEAD] [DCO] built on Feb  6 2023
2023-02-15 09:55:41 Windows version 10.0 (Windows 10 or greater), amd64 executable
2023-02-15 09:55:41 library versions: OpenSSL 3.0.7 1 Nov 2022, LZO 2.10
2023-02-15 09:55:41 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:25341
2023-02-15 09:55:41 Need hold release from management interface, waiting...
2023-02-15 09:55:41 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:53761
2023-02-15 09:55:41 MANAGEMENT: CMD 'state on'
2023-02-15 09:55:41 MANAGEMENT: CMD 'log on all'
2023-02-15 09:55:41 MANAGEMENT: CMD 'echo on all'
2023-02-15 09:55:41 MANAGEMENT: CMD 'bytecount 5'
2023-02-15 09:55:41 MANAGEMENT: CMD 'state'
2023-02-15 09:55:41 MANAGEMENT: CMD 'hold off'
2023-02-15 09:55:41 MANAGEMENT: CMD 'hold release'
2023-02-15 09:55:46 MANAGEMENT: CMD 'password [...]'
2023-02-15 09:55:46 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
2023-02-15 09:55:46 Outgoing Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:55:46 Incoming Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:55:46 TCP/UDP: Preserving recently used remote address: [AF_INET]xxxxxxxxxxxxxxxxxxx:443
2023-02-15 09:55:46 ovpn-dco device [OpenVPN Data Channel Offload] opened
2023-02-15 09:55:46 TCP_CLIENT link local: (not bound)
2023-02-15 09:55:46 TCP_CLIENT link remote: [AF_INET]xxxxxxxxxxxxxxxxxxx:443
2023-02-15 09:55:46 MANAGEMENT: >STATE:1676451346,WAIT,,,,,,
2023-02-15 09:55:46 MANAGEMENT: >STATE:1676451346,AUTH,,,,,,
2023-02-15 09:55:46 TLS: Initial packet from [AF_INET]xxxxxxxxxxxxxx:443, sid=9c7a3b70 c40f23cc
2023-02-15 09:55:46 VERIFY OK: depth=2, xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, CN=RootCA, emailAddress=xxxxxxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:46 VERIFY OK: depth=1, xxxxxxxxxxxxxxxxxxxxxxxxxx CN=ServerIntermediateCA, emailAddress=xxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:46 VERIFY KU OK
2023-02-15 09:55:46 Validating certificate extended key usage
2023-02-15 09:55:46 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2023-02-15 09:55:46 VERIFY EKU OK
2023-02-15 09:55:46 VERIFY OK: depth=0, xxxxxxxxxxxxxxxxxxxxxxxxxxx CN=OpenVPN1-server, emailAddress=xxxxxxxxxxxxx
2023-02-15 09:55:46 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 2048 bit RSA, signature: RSA-SHA256
2023-02-15 09:55:46 [OpenVPN1-server] Peer Connection Initiated with [AF_INET]xxxxxxxxxxxxxxxxxxxxx:443
2023-02-15 09:55:46 TLS: move_session: dest=TM_ACTIVE src=TM_INITIAL reinit_src=1
2023-02-15 09:55:46 TLS: tls_multi_process: initial untrusted session promoted to trusted
2023-02-15 09:55:46 PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS xxxxxxxxxxxxxxxxxx,dhcp-option DNS xxxxxxxxxxxx,dhcp-option DOMAIN xxxxxxxxxxxxxxxxxx,route 1xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:46 OPTIONS IMPORT: timers and/or timeouts modified
2023-02-15 09:55:46 OPTIONS IMPORT: --ifconfig/up options modified
2023-02-15 09:55:46 OPTIONS IMPORT: route options modified
2023-02-15 09:55:46 OPTIONS IMPORT: route-related options modified
2023-02-15 09:55:46 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
2023-02-15 09:55:46 OPTIONS IMPORT: peer-id set
2023-02-15 09:55:46 OPTIONS IMPORT: data channel crypto options modified
2023-02-15 09:55:46 interactive service msg_channel=868
2023-02-15 09:55:46 MANAGEMENT: >STATE:1676451346,ASSIGN_IP,,10.8.1.2,,,,
2023-02-15 09:55:46 INET address service: add 10.8.1.2/24
2023-02-15 09:55:46 IPv4 dns servers set using service
2023-02-15 09:55:46 DNS domain set using service
2023-02-15 09:55:46 IPv4 MTU set to 1500 on interface 61 using service
2023-02-15 09:55:46 MANAGEMENT: >STATE:1676451346,ADD_ROUTES,,,,,,
2023-02-15 09:55:46 C:\windows\system32\route.exe ADD xxxxxx MASK xxxxxx xxxxxxx METRIC 200
2023-02-15 09:55:46 Route addition via service succeeded
2023-02-15 09:55:46 Initialization Sequence Completed
2023-02-15 09:55:46 MANAGEMENT: >STATE:1676451346,CONNECTED,SUCCESS,10.8.1.2,10.210.152.3,443,,
2023-02-15 09:55:54 Connection reset, restarting [-1]
2023-02-15 09:55:54 C:\windows\system32\route.exe DELETE xxxxxxxxxxx MASK 255.255.0.0 xxxxxxxxxxxxxxxxx
2023-02-15 09:55:54 Route deletion via service succeeded
2023-02-15 09:55:54 Closing DCO interface
2023-02-15 09:55:54 DNS domain deleted using service
2023-02-15 09:55:54 IPv4 dns servers deleted using service
2023-02-15 09:55:54 INET address service: remove 10.8.1.2/24
2023-02-15 09:55:54 SIGUSR1[soft,connection-reset] received, process restarting
2023-02-15 09:55:54 MANAGEMENT: >STATE:1676451354,RECONNECTING,connection-reset,,,,,
2023-02-15 09:55:54 Restart pause, 1 second(s)
2023-02-15 09:55:55 Outgoing Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:55:55 Incoming Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:55:55 TCP/UDP: Preserving recently used remote address: [AF_INET]10.210.152.3:443
2023-02-15 09:55:55 ovpn-dco device [OpenVPN Data Channel Offload] opened
2023-02-15 09:55:55 TCP_CLIENT link local: (not bound)
2023-02-15 09:55:55 TCP_CLIENT link remote: [AF_INET]10.210.152.3:443
2023-02-15 09:55:55 MANAGEMENT: >STATE:1676451355,WAIT,,,,,,
2023-02-15 09:55:55 MANAGEMENT: >STATE:1676451355,AUTH,,,,,,
2023-02-15 09:55:55 TLS: Initial packet from [AF_INET]10.210.152.3:443, sid=34d8f401 fe9bab6e
2023-02-15 09:55:55 VERIFY OK: depth=2, xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx CN=RootCA, emailAddress=xxxxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:55 VERIFY OK: depth=1, xxxxxxxxxxxxxxxxxxxxxxxxx CN=ServerIntermediateCA, emailAddress=xxxxxxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:55 VERIFY KU OK
2023-02-15 09:55:55 Validating certificate extended key usage
2023-02-15 09:55:55 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2023-02-15 09:55:55 VERIFY EKU OK
2023-02-15 09:55:55 VERIFY OK: depth=0, xxxxxxxxxxxxxxxxxxxxxxxxx CN=OpenVPN1-server, emailAddress=xxxxxxxxxxxxxxxxxxxxxxxxxx
2023-02-15 09:55:55 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 2048 bit RSA, signature: RSA-SHA256
2023-02-15 09:55:55 [OpenVPN1-server] Peer Connection Initiated with [AF_INET]xxxxxxxxxxxxxxxx:443
2023-02-15 09:55:55 TLS: move_session: dest=TM_ACTIVE src=TM_INITIAL reinit_src=1
2023-02-15 09:55:55 TLS: tls_multi_process: initial untrusted session promoted to trusted
2023-02-15 09:55:56 MANAGEMENT: >STATE:1676451356,GET_CONFIG,,,,,,
2023-02-15 09:55:56 SENT CONTROL [OpenVPN1-server]: 'PUSH_REQUEST' (status=1)
2023-02-15 09:55:57 AUTH: Received control message: AUTH_FAILED
2023-02-15 09:55:57 Closing DCO interface
2023-02-15 09:55:57 SIGUSR1[soft,auth-failure] received, process restarting
2023-02-15 09:55:57 MANAGEMENT: >STATE:1676451357,RECONNECTING,auth-failure,,,,,
2023-02-15 09:55:57 Restart pause, 1 second(s)
2023-02-15 09:56:30 MANAGEMENT: CMD 'password [...]'
2023-02-15 09:56:30 OpenSSL: error:1C800064:Provider routines::bad decrypt
2023-02-15 09:56:30 OpenSSL: error:11800074:PKCS12 routines::pkcs12 cipherfinal error
2023-02-15 09:56:30 OpenSSL: error:1C800064:Provider routines::bad decrypt
2023-02-15 09:56:30 OpenSSL: error:11800074:PKCS12 routines::pkcs12 cipherfinal error
2023-02-15 09:56:30 Cannot load private key file gphilippon.key
2023-02-15 09:56:30 SIGUSR1[soft,private-key-password-failure] received, process restarting
2023-02-15 09:56:30 MANAGEMENT: >STATE:1676451390,RECONNECTING,private-key-password-failure,,,,,
2023-02-15 09:56:30 Restart pause, 1 second(s)
2023-02-15 09:56:52 MANAGEMENT: CMD 'password [...]'
2023-02-15 09:56:52 Outgoing Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:56:52 Incoming Control Channel Authentication: Using 256 bit message hash 'SHA256' for HMAC authentication
2023-02-15 09:56:52 TCP/UDP: Preserving recently used remote address: [AF_INET]xxxxxxxxxxxxxxxxxxx:443
2023-02-15 09:56:52 ovpn-dco device [OpenVPN Data Channel Offload] opened
2023-02-15 09:56:52 TCP_CLIENT link local: (not bound)
2023-02-15 09:56:52 TCP_CLIENT link remote: [AF_INET]xxxxxxxxxxxxxxxxxx:443
2023-02-15 09:56:52 MANAGEMENT: >STATE:1676451412,WAIT,,,,,,
2023-02-15 09:56:52 MANAGEMENT: >STATE:1676451412,AUTH,,,,,,
2023-02-15 09:56:52 TLS: Initial packet from [AF_INET]xxxxxxxxxxxxxxxxxxxx:443, sid=5211d5ef 90765c65
2023-02-15 09:56:52 VERIFY OK: depth=2, xxxxxxxxxxxxxxxxxxxxx, CN=RootCA, emailAddress=xxxxxxxxxxxxxxx
2023-02-15 09:56:52 VERIFY OK: depth=1, xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, CN=ServerIntermediateCA, emailAddress=xxxxxxxxxxxxxxxxxx
2023-02-15 09:56:52 VERIFY KU OK
2023-02-15 09:56:52 Validating certificate extended key usage
2023-02-15 09:56:52 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2023-02-15 09:56:52 VERIFY EKU OK
2023-02-15 09:56:52 VERIFY OK: depth=0, xxxxxxxxxxxxxxxxxxxxxxxxxx, CN=OpenVPN-server, emailAddress=xxxxxxxxxxxxxxxxxxx
2023-02-15 09:56:52 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 2048 bit RSA, signature: RSA-SHA256
2023-02-15 09:56:52 [OpenVPN-server] Peer Connection Initiated with [AF_INET]xxxxxxxxxxxxxxxx:443
2023-02-15 09:56:52 TLS: move_session: dest=TM_ACTIVE src=TM_INITIAL reinit_src=1
2023-02-15 09:56:52 TLS: tls_multi_process: initial untrusted session promoted to trusted
2023-02-15 09:56:53 MANAGEMENT: >STATE:1676451413,GET_CONFIG,,,,,,
2023-02-15 09:56:53 SENT CONTROL [OpenVPN-server]: 'PUSH_REQUEST' (status=1)
2023-02-15 09:56:54 AUTH: Received control message: AUTH_FAILED
2023-02-15 09:56:54 Closing DCO interface
2023-02-15 09:56:54 SIGUSR1[soft,auth-failure] received, process restarting
2023-02-15 09:56:54 MANAGEMENT: >STATE:1676451414,RECONNECTING,auth-failure,,,,,
2023-02-15 09:56:54 Restart pause, 1 second(s)
`

I've also tried by deactivating the adapter Data Channel Offload, unticked the DNE Light Weight Filter on both adapter Data Channel Offload and OpenVPN Tap Windows6 but the connection is still not working after a few sec.

**Comment by schwabe** [comment]
`2023-02-15 09:56:54 AUTH: Received control message: AUTH_FAILED`

The server is rejecting you.

**Comment by Guigz25** [comment]
I've checked the log file on the server side and it appears that my problem is linked to the MTU problem.
WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1551', remote='link-mtu 1554'
The problem is I've specified the opt-verify  which reject clients having bad configuration. 

I've tried setting the mtu on the client conf 
link-mtu 1554

But I have another error with tun-mtu. 

After reading the changelog of the new version, I can see they were numerous changes concerning mtu. 
max-packet-size
tun-mtu-max

I've tried putting these options and setting different values but the problem still remains. 

I'm gessing that commenting the opt-verify option will fix the problem but I can't do it right now because I have users connected to the VPN.

**Comment by schwabe** [comment]
Yes. `opt-verify` is deprecated and should not be used anymore. The OCC check is ill equipped to deal with negotiations and often causes problems like you are seeing here.

**Comment by Guigz25** [comment]
Ok I have 3 OpenVPN serveur and one of them was unused so I can confirm that everything is working. 
So the solution consisted to manually unticked the DNE LightWeight Filter on the DCO adapter. 
Do you have any plan to fix this bug in the next release ? 

Moreover, I still have a warning concerning the use of tls-groups instead of a specific curve.
I'm actually using this curve : brainpoolP256r1
Can you tell me which tls-groups is the most secure ? 
I have follow the best pratices established by ANSSI but they were using this specific ecdh-curves.

**Comment by cron2** [comment]
Hi,

On Wed, Feb 15, 2023 at 02:06:22AM -0800, Guigz25 wrote:
> I'm gessing that commenting the opt-verify option will fix the problem but I can't do it right now because I have users connected to the VPN. 

opt-verify is an option from the dark ages when OpenVPN had no real
client/server option handshaking.  This should no longer be used
(if you run the server on 2.6.0, all these MTU things will be ignored
for option checking, but we can't retroactively teach old and existing
servers that "things have changed").

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by lstipakov** [comment]
> Ok I have 3 OpenVPN serveur and one of them was unused so I can confirm that everything is working. So the solution consisted to manually unticked the DNE LightWeight Filter on the DCO adapter. Do you have any plan to fix this bug in the next release ?

Yes, this will be fixed in 2.6.0-I005.

**Comment by Guigz25** [comment]
Do you have an ETA for the fix ? 
And do you have any recommandation concerning the use of specific tls-groups ? I really would like to get rid of this warning and keeping the maximum security (so without commenting the ecdh-curves)

**Comment by schwabe** [comment]
The warning of `ecdh-curve` vs `tls-groups` is just that with modern OpenSSL and TLS implementations, you can specify a list of eliptic curves/groups and the new option will now allow this. And as the man page says, the default with OpenSSL 1.1 is "X25519:secp256r1:X448:secp521r1:secp384r1". Some of these curves, especially when using OpenSSL 3/TLS 1.3 are kind of mandatory. Not including X25519 iirc breaks TLS 1.3 support. What set of curves/groups is acceptable ANSSI, I have no idea.  But as far as I am aware, brainpoolP256r1 is not specified for the use with TLS 1.3

**Comment by Guigz25** [comment]
Perfect ! 
Thank you @schwabe ;)

**Comment by lstipakov** [comment]
Citrix DNE issue is fixed in 2.6.0-I005.
